src/poly/commitment.rs

```diff
@@ -25,21 +27,13 @@ pub struct Params<C: CurveAffine> {
     pub(crate) g: Vec<C>,
     pub(crate) g_lagrange: Vec<C>,
     pub(crate) h: C,
-}
-
-/// This is a proof object for the polynomial commitment scheme opening.
-#[derive(Debug, Clone)]
-pub struct Proof<C: CurveAffine> {
-    rounds: Vec<(C, C)>,
-    delta: C,
-    z1: C::Scalar,
-    z2: C::Scalar,
+    pub(crate) u: C,
 }
 
 impl<C: CurveAffine> Params<C> {
     /// Initializes parameters for the curve, given a random oracle to draw
     /// points from.
-    pub fn new<H: Hasher<C::Base>>(k: u32) -> Self {
+    pub fn new<T: TranscriptWrite<io::Sink, C> + Sync>(k: u32) -> Self {
```

Comment:
In order to remove the writer type parameter we'll have to abandon the (misuse) of `TranscriptWrite` in this API. I'll just bring in `blake2b` and we'll sample points for the parameters with that instead until we have a better plan.